throttler/throttler.go

```diff
@@ -0,0 +1,78 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+//ThrottledAction is a common interface for actions that can be executed
+type ThrottledAction interface {
+	Execute()
+}
```

Comment:
It makes sense to couple this package to other packages? 
Also, the `server.Runner` objective is... running servers, no? What if we want to throttle "one-shot" actions and not long running procedures?

Even that, maybe it makes sense to add the `context.Context` as an argument to the interface?